sdkgen/nodejs-request/lib/index.js

```diff
@@ -0,0 +1,73 @@
+// TODO add options and fetch options
+const processCollection = require('../../../lib/utils').processCollection,
+  { sanitize, itemGroupHandler, itemHandler} = require('./util');
+
+/**
+ * Generates sdk for nodejs-request
+
+ * @param {PostmanCollection} collection - Postman collection Instance
+ * @param {Object} options - postman-sdk-generators options
+ * @param {Object} [options.variableList] - variableList to be declared in generated sdk
+ * @param {Function} callback - callback functio to return results (err, response)
+ * @returns {String} - sdk snippet for input collection
+ * TODO add indentation to entire snippet
+ */
+async function generate (collection, options, callback) {
+  var snippet = '',
+    indent = options.indentType === 'Tab' ? '\t' : ' ';
+  indent = indent.repeat(options.indentCount);
+
+  if (options.ES6_enabled) {
+    snippet += 'const ';
+  }
+  else {
+    snippet += 'var ';
+  }
+  snippet += 'request = require(\'request\');\n\n';
+  snippet += indent + 'const configVariables = {';
+  if (options.variableList) {
+    options.variableList.each((item) => {
+      snippet += indent.repeat(2) + `'${sanitize(item.key)}': '${sanitize(item.value)}',\n`;
+    });
+  }
+  snippet += indent + '};\n\n';
+  snippet += 'function SDK(environment = {}) {\n\n';
+  snippet += options.ES6_enabled ? 'const ' : 'var ';
+  snippet += 'self = this;\n\n';
+  snippet += indent + 'this.requests = {\n';
+  try {
+    snippet += await processCollection(collection, options, itemHandler, itemGroupHandler);
+  }
+  catch (err) {
+    return callback(err, null);
+  }
+  snippet += indent + `}.${collection.name};\n\n`;
+  snippet += indent + 'this.variables = this.setVariables(environment);\n\n';
+  snippet += '}\n\n';
+  snippet += '/**\n';
+  snippet += 'Function to set environment variables. These variables will override the collection variables\n\n';
```

Comment:
Function to set ~environment~ variables. These variables will override the **existing/default variables values**.